4.8 Tasks

In [1]:
#importing libraries
import pandas as pd
import numpy as np
import os

In [3]:
#importing data files
path = r'/Users/tiffanib./Desktop/Career Foundry Data Analytics/Achievement 4 - Python/28-09-2025 Instacart Basket Analysis'

In [4]:
path

'/Users/tiffanib./Desktop/Career Foundry Data Analytics/Achievement 4 - Python/28-09-2025 Instacart Basket Analysis'

Number 1. Create a new notebook for this task. Be sure to import the relevant libraries, along with your ords_prods_merge dataframe, which should include your newly derived columns from the previous Exercise.

In [10]:
ords_prods_merged_NV = pd.read_pickle(os.path.join(path, '02 Data' , 'Prepared Data' , 'ords_prods_merged_NV.pkl'))

In [11]:
ords_prods_merged_NV.shape

(32404859, 22)

In [13]:
ords_prods_merged_NV.columns

Index(['Unnamed: 0.1', 'Unnamed: 0', 'order_id', 'user_id', 'order_number',
       'orders_day_of_week', 'orders_hour_of_day', 'days_since_prior_order',
       'product_id', 'add_to_cart_order', 'reordered', '_merge',
       'product_name', 'aisle_id', 'department_id', 'prices', 'match',
       'prices_range_loc', 'price_range_loc', 'busiest_day', 'busiest_days',
       'busiest_period_of_day'],
      dtype='object')

In [15]:
#Checking the user Id column to make sure everything imported correctly
ords_prods_merged_NV['user_id']

0                1
1                1
2                1
3                1
4                1
             ...  
32404854    206209
32404855    206209
32404856    206209
32404857    206209
32404858    206209
Name: user_id, Length: 32404859, dtype: int64

Number 2. In this Exercise, you learned how to find the aggregated mean of the “order_number” column grouped by “department_id” for a subset of your dataframe. Now, repeat this process for the entire dataframe.

In [17]:
#grouping by department_id and order number to caluculate the mean using agg() function
ords_prods_merged_NV.groupby('department_id').agg({'order_number' :{'mean'}})

order_number
                      mean
department_id             
1                15.457838
2                17.277920
3                17.170395
4                17.811403
5                15.215751
6                16.439806
7                17.225802
8                15.340650
9                15.895474
10               20.197148
11               16.170638
12               15.887671
13               16.583536
14               16.773669
15               16.165037
16               17.665606
17               15.694469
18               19.310397
19               17.177343
20               16.473447
21               22.902379

Number 3. Analyze the result. How do the results for the entire dataframe differ from those of the subset? Include your comments in a markdown cell below the executed code.

the full dataframe shows higher average order counts across some departments suggesting it could reflect a wider spectrum of customer purchasing patterns than the subset.
Department 21 shows the highest mean in both datasets but in this instance it is higher than the subset (21.956893). Department 1 shows a higher mean in the dataframe at 15.457838 than in the subset 14.800024.

Number 4. Follow the instructions in the Exercise for creating a loyalty flag for existing customers using the transform() and loc() functions.

In [19]:
ords_prods_merged_NV['max_order'] = ords_prods_merged_NV.groupby(['user_id'])['order_number'].transform('max')

In [20]:
#changed code due to error in system. Used 'max' instead of np.max
#checking output
ords_prods_merged_NV.head(22)

Unnamed: 0.1  Unnamed: 0  order_id  user_id  order_number  \
0              0           0   2539329        1             1   
1              0           0   2539329        1             1   
2              0           0   2539329        1             1   
3              0           0   2539329        1             1   
4              0           0   2539329        1             1   
5              1           1   2398795        1             2   
6              1           1   2398795        1             2   
7              1           1   2398795        1             2   
8              1           1   2398795        1             2   
9              1           1   2398795        1             2   
10             1           1   2398795        1             2   
11             2           2    473747        1             3   
12             2           2    473747        1             3   
13             2           2    473747        1             3   
14             2           2    473747        1             3   
15             2           2    473747        1             3   
16             3           3   2254736        1             4   
17             3           3   2254736        1             4   
18             3           3   2254736        1             4   
19             3           3   2254736        1             4   
20             3           3   2254736        1             4   
21             4           4    431534        1             5   

    orders_day_of_week  orders_hour_of_day  days_since_prior_order  \
0                    2                   8                     NaN   
1                    2                   8                     NaN   
2                    2                   8                     NaN   
3                    2                   8                     NaN   
4                    2                   8                     NaN   
5                    3                   7                    15.0   
6                    3                   7                    15.0   
7                    3                   7                    15.0   
8                    3                   7                    15.0   
9                    3                   7                    15.0   
10                   3                   7                    15.0   
11                   3                  12                    21.0   
12                   3                  12                    21.0   
13                   3                  12                    21.0   
14                   3                  12                    21.0   
15                   3                  12                    21.0   
16                   4                   7                    29.0   
17                   4                   7                    29.0   
18                   4                   7                    29.0   
19                   4                   7                    29.0   
20                   4                   7                    29.0   
21                   4                  15                    28.0   

    product_id  add_to_cart_order  ...  aisle_id department_id prices  match  \
0          196                  1  ...        77             7    9.0   both   
1        14084                  2  ...        91            16   12.5   both   
2        12427                  3  ...        23            19    4.4   both   
3        26088                  4  ...        23            19    4.7   both   
4        26405                  5  ...        54            17    1.0   both   
5          196                  1  ...        77             7    9.0   both   
6        10258                  2  ...       117            19    3.0   both   
7        12427                  3  ...        23            19    4.4   both   
8        13176                  4  ...        24             4   10.3   both   
9        26088                  5  ...        23            19    4.7   both   
10       13032   

In [22]:
ords_prods_merged_NV.loc[ords_prods_merged_NV['max_order'] > 40, 'loyalty_flag'] = 'Loyal customer'
ords_prods_merged_NV.loc[(ords_prods_merged_NV['max_order'] <= 40) & (ords_prods_merged_NV['max_order'] > 10), 'loyalty_flag'] = 'Regular customer'
ords_prods_merged_NV.loc[ords_prods_merged_NV['max_order'] <= 10, 'loyalty_flag'] = 'New customer'

In [23]:
#checking the value count on the loyalty flag
ords_prods_merged_NV['loyalty_flag'].value_counts(dropna = False)

loyalty_flag
Regular customer    15876776
Loyal customer      10284093
New customer         6243990
Name: count, dtype: int64

In [24]:
#checking the updated output
ords_prods_merged_NV.head(25)

Unnamed: 0.1  Unnamed: 0  order_id  user_id  order_number  \
0              0           0   2539329        1             1   
1              0           0   2539329        1             1   
2              0           0   2539329        1             1   
3              0           0   2539329        1             1   
4              0           0   2539329        1             1   
5              1           1   2398795        1             2   
6              1           1   2398795        1             2   
7              1           1   2398795        1             2   
8              1           1   2398795        1             2   
9              1           1   2398795        1             2   
10             1           1   2398795        1             2   
11             2           2    473747        1             3   
12             2           2    473747        1             3   
13             2           2    473747        1             3   
14             2           2    473747        1             3   
15             2           2    473747        1             3   
16             3           3   2254736        1             4   
17             3           3   2254736        1             4   
18             3           3   2254736        1             4   
19             3           3   2254736        1             4   
20             3           3   2254736        1             4   
21             4           4    431534        1             5   
22             4           4    431534        1             5   
23             4           4    431534        1             5   
24             4           4    431534        1             5   

    orders_day_of_week  orders_hour_of_day  days_since_prior_order  \
0                    2                   8                     NaN   
1                    2                   8                     NaN   
2                    2                   8                     NaN   
3                    2                   8                     NaN   
4                    2                   8                     NaN   
5                    3                   7                    15.0   
6                    3                   7                    15.0   
7                    3                   7                    15.0   
8                    3                   7                    15.0   
9                    3                   7                    15.0   
10                   3                   7                    15.0   
11                   3                  12                    21.0   
12                   3                  12                    21.0   
13                   3                  12                    21.0   
14                   3                  12                    21.0   
15                   3                  12                    21.0   
16                   4                   7                    29.0   
17                   4                   7                    29.0   
18                   4                   7                    29.0   
19                   4                   7                    29.0   
20                   4                   7                    29.0   
21                   4                  15                    28.0   
22                   4                  15                    28.0   
23                   4                  15                    28.0   
24                   4                  15                    28.0   

    product_id  add_to_cart_order  ...  department_id prices match  \
0          196                  1  ...              7    9.0  both   
1        14084                  2  ...             16   12.5  both   
2        12427                  3  ...             19    4.4  both   
3        26088                  4  ...             19    4.7  both   
4        26405                  5  ...             17    1.0  both   
5          196                  1  ...              7    9.0  both   
6 

In [25]:
ords_prods_merged_NV.tail(25)

Unnamed: 0.1  Unnamed: 0  order_id  user_id  order_number  \
32404834       3421080     3421080    626363   206209            12   
32404835       3421080     3421080    626363   206209            12   
32404836       3421080     3421080    626363   206209            12   
32404837       3421080     3421080    626363   206209            12   
32404838       3421080     3421080    626363   206209            12   
32404839       3421080     3421080    626363   206209            12   
32404840       3421080     3421080    626363   206209            12   
32404841       3421080     3421080    626363   206209            12   
32404842       3421080     3421080    626363   206209            12   
32404843       3421080     3421080    626363   206209            12   
32404844       3421080     3421080    626363   206209            12   
32404845       3421080     3421080    626363   206209            12   
32404846       3421080     3421080    626363   206209            12   
32404847       3421080     3421080    626363   206209            12   
32404848       3421080     3421080    626363   206209            12   
32404849       3421080     3421080    626363   206209            12   
32404850       3421081     3421081   2977660   206209            13   
32404851       3421081     3421081   2977660   206209            13   
32404852       3421081     3421081   2977660   206209            13   
32404853       3421081     3421081   2977660   206209            13   
32404854       3421081     3421081   2977660   206209            13   
32404855       3421081     3421081   2977660   206209            13   
32404856       3421081     3421081   2977660   206209            13   
32404857       3421081     3421081   2977660   206209            13   
32404858       3421081     3421081   2977660   206209            13   

          orders_day_of_week  orders_hour_of_day  days_since_prior_order  \
32404834                   1                  12                    18.0   
32404835                   1                  12                    18.0   
32404836                   1                  12                    18.0   
32404837                   1                  12                    18.0   
32404838                   1                  12                    18.0   
32404839                   1                  12                    18.0   
32404840                   1                  12                    18.0   
32404841                   1                  12                    18.0   
32404842                   1                  12                    18.0   
32404843                   1                  12                    18.0   
32404844                   1                  12                    18.0   
32404845                   1                  12                    18.0   
32404846                   1                  12                    18.0   
32404847                   1                  12                    18.0   
32404848                   1                  12                    18.0   
32404849                   1                  12                    18.0   
32404850                   1                  12                     7.0   
32404851                   1                  12                     7.0   
32404852                   1                  12                     7.0   
32404853                   1                  12                     7.0   
32404854                   1                  12                     7.0   
32404855                   1                  12                     7.0   
32404856                   1                  12                     7.0   
32404857                   1                  12                     7.0   
32404858                   1                  12                     7.0   

          product_id  add_to_cart_order  ...  department_id prices match  \
32404834       41213                  5  ...              3    4.8  both   
32404835       15700              

Number 5. The marketing team at Instacart wants to know whether there’s a difference between the spending habits of the three types of customers you identified. Use the loyalty flag you created and check the basic statistics of the product prices for each loyalty category (Loyal Customer, Regular Customer, and New Customer). What you’re trying to determine is whether the prices of products purchased by loyal customers differ from those purchased by regular or new customers.

In [27]:
ords_prods_merged_NV.groupby('loyalty_flag').agg({'prices' :['mean', 'min', 'max']})

prices              
                       mean  min      max
loyalty_flag                             
Loyal customer    10.386336  1.0  99999.0
New customer      13.294670  1.0  99999.0
Regular customer  12.495717  1.0  99999.0

The loyal customers tend to purchase lower cost items at 10.39, regular customers are in the middle with an average of 12.50, and the new customers spend the most with 13.29. The price of 99999 could be an outlier or data errors.

Number 6. The team now wants to target different types of spenders in their marketing campaigns. This can be achieved by looking at the prices of the items people are buying. Create a spending flag for each user based on the average price across all their orders using the following criteria:

If the mean of the prices of products purchased by a user is lower than 10, then flag them as a “Low spender.”

If the mean of the prices of products purchased by a user is higher than or equal to 10, then flag them as a “High spender.”

In [30]:
ords_prods_merged_NV['avg_product_price'] = ords_prods_merged_NV.groupby(['user_id'])['prices'].transform('mean')

In [31]:
ords_prods_merged_NV [['user_id','avg_product_price']].head(20)

user_id  avg_product_price
0         1           6.367797
1         1           6.367797
2         1           6.367797
3         1           6.367797
4         1           6.367797
5         1           6.367797
6         1           6.367797
7         1           6.367797
8         1           6.367797
9         1           6.367797
10        1           6.367797
11        1           6.367797
12        1           6.367797
13        1           6.367797
14        1           6.367797
15        1           6.367797
16        1           6.367797
17        1           6.367797
18        1           6.367797
19        1           6.367797

In [32]:
ords_prods_merged_NV [['user_id','avg_product_price']].tail(20)

user_id  avg_product_price
32404839   206209           7.058915
32404840   206209           7.058915
32404841   206209           7.058915
32404842   206209           7.058915
32404843   206209           7.058915
32404844   206209           7.058915
32404845   206209           7.058915
32404846   206209           7.058915
32404847   206209           7.058915
32404848   206209           7.058915
32404849   206209           7.058915
32404850   206209           7.058915
32404851   206209           7.058915
32404852   206209           7.058915
32404853   206209           7.058915
32404854   206209           7.058915
32404855   206209           7.058915
32404856   206209           7.058915
32404857   206209           7.058915
32404858   206209           7.058915

In [35]:
# Creating a 'Spending' Label Flag using loc Function
ords_prods_merged_NV.loc[ords_prods_merged_NV['avg_product_price'] >= 10, 'spending_flag'] = 'High spender'
ords_prods_merged_NV.loc[ords_prods_merged_NV['avg_product_price'] < 10, 'spending_flag'] = 'Low spender'

In [36]:
#checking the value count on the spending flag column
ords_prods_merged_NV['spending_flag'].value_counts(dropna = False)

spending_flag
Low spender     31770614
High spender      634245
Name: count, dtype: int64

In [37]:
ords_prods_merged_NV[['user_id', 'avg_product_price', 'spending_flag']].head(15)

user_id  avg_product_price spending_flag
0         1           6.367797   Low spender
1         1           6.367797   Low spender
2         1           6.367797   Low spender
3         1           6.367797   Low spender
4         1           6.367797   Low spender
5         1           6.367797   Low spender
6         1           6.367797   Low spender
7         1           6.367797   Low spender
8         1           6.367797   Low spender
9         1           6.367797   Low spender
10        1           6.367797   Low spender
11        1           6.367797   Low spender
12        1           6.367797   Low spender
13        1           6.367797   Low spender
14        1           6.367797   Low spender

In [38]:
ords_prods_merged_NV[['user_id', 'avg_product_price', 'spending_flag']].tail(15)

user_id  avg_product_price spending_flag
32404844   206209           7.058915   Low spender
32404845   206209           7.058915   Low spender
32404846   206209           7.058915   Low spender
32404847   206209           7.058915   Low spender
32404848   206209           7.058915   Low spender
32404849   206209           7.058915   Low spender
32404850   206209           7.058915   Low spender
32404851   206209           7.058915   Low spender
32404852   206209           7.058915   Low spender
32404853   206209           7.058915   Low spender
32404854   206209           7.058915   Low spender
32404855   206209           7.058915   Low spender
32404856   206209           7.058915   Low spender
32404857   206209           7.058915   Low spender
32404858   206209           7.058915   Low spender

Number 7. In order to send relevant notifications to users within the app (for instance, asking users if they want to buy the same item again), the Instacart team wants you to determine frequent versus non-frequent customers. Create an order frequency flag that marks the regularity of a user’s ordering behavior according to the median in the “days_since_prior_order” column. The criteria for the flag should be as follows:

-If the median of “days_since_prior_order” is higher than 20, then the customer should be labeled a “Non-frequent customer.”

-If the median is higher than 10 and lower than or equal to 20, then the customer should be labeled a “Regular customer.”

-If the median is lower than or equal to 10, then the customer should be labeled a “Frequent customer.”

In [39]:
ords_prods_merged_NV['median_days_between_orders'] = ords_prods_merged_NV.groupby(['user_id'])['days_since_prior_order'].transform('median')

In [40]:
#checking output
ords_prods_merged_NV[['user_id', 'days_since_prior_order','median_days_between_orders']].head(17)

user_id  days_since_prior_order  median_days_between_orders
0         1                     NaN                        20.5
1         1                     NaN                        20.5
2         1                     NaN                        20.5
3         1                     NaN                        20.5
4         1                     NaN                        20.5
5         1                    15.0                        20.5
6         1                    15.0                        20.5
7         1                    15.0                        20.5
8         1                    15.0                        20.5
9         1                    15.0                        20.5
10        1                    15.0                        20.5
11        1                    21.0                        20.5
12        1                    21.0                        20.5
13        1                    21.0                        20.5
14        1                    21.0                        20.5
15        1                    21.0                        20.5
16        1                    29.0                        20.5

In [41]:
ords_prods_merged_NV[['user_id', 'days_since_prior_order','median_days_between_orders']].tail(17)

user_id  days_since_prior_order  median_days_between_orders
32404842   206209                    18.0                        22.0
32404843   206209                    18.0                        22.0
32404844   206209                    18.0                        22.0
32404845   206209                    18.0                        22.0
32404846   206209                    18.0                        22.0
32404847   206209                    18.0                        22.0
32404848   206209                    18.0                        22.0
32404849   206209                    18.0                        22.0
32404850   206209                     7.0                        22.0
32404851   206209                     7.0                        22.0
32404852   206209                     7.0                        22.0
32404853   206209                     7.0                        22.0
32404854   206209                     7.0                        22.0
32404855   206209                     7.0                        22.0
32404856   206209                     7.0                        22.0
32404857   206209                     7.0                        22.0
32404858   206209                     7.0                        22.0

In [42]:
#creating an order frequncy label flag
ords_prods_merged_NV.loc[ords_prods_merged_NV['median_days_between_orders'] > 20, 'order_frequency_flag'] = 'Non-frequent customer'
ords_prods_merged_NV.loc[(ords_prods_merged_NV['median_days_between_orders'] <= 20) & (ords_prods_merged_NV['median_days_between_orders'] > 10), 'order_frequency_flag'] = 'Regular customer'
ords_prods_merged_NV.loc[ords_prods_merged_NV['median_days_between_orders'] <= 10, 'order_frequency_flag'] = 'Frequent customer'

In [43]:
#checking output of value counts on the order frequency label flag
ords_prods_merged_NV ['order_frequency_flag'].value_counts(dropna = False)

order_frequency_flag
Frequent customer        21559853
Regular customer          7208564
Non-frequent customer     3636437
NaN                             5
Name: count, dtype: int64

In [45]:
#checking dataframes created to ensure they have updated
ords_prods_merged_NV [['user_id', 'days_since_prior_order','median_days_between_orders','order_frequency_flag',]].head(20)

user_id  days_since_prior_order  median_days_between_orders  \
0         1                     NaN                        20.5   
1         1                     NaN                        20.5   
2         1                     NaN                        20.5   
3         1                     NaN                        20.5   
4         1                     NaN                        20.5   
5         1                    15.0                        20.5   
6         1                    15.0                        20.5   
7         1                    15.0                        20.5   
8         1                    15.0                        20.5   
9         1                    15.0                        20.5   
10        1                    15.0                        20.5   
11        1                    21.0                        20.5   
12        1                    21.0                        20.5   
13        1                    21.0                        20.5   
14        1                    21.0                        20.5   
15        1                    21.0                        20.5   
16        1                    29.0                        20.5   
17        1                    29.0                        20.5   
18        1                    29.0                        20.5   
19        1                    29.0                        20.5   

     order_frequency_flag  
0   Non-frequent customer  
1   Non-frequent customer  
2   Non-frequent customer  
3   Non-frequent customer  
4   Non-frequent customer  
5   Non-frequent customer  
6   Non-frequent customer  
7   Non-frequent customer  
8   Non-frequent customer  
9   Non-frequent customer  
10  Non-frequent customer  
11  Non-frequent customer  
12  Non-frequent customer  
13  Non-frequent customer  
14  Non-frequent customer  
15  Non-frequent customer  
16  Non-frequent customer  
17  Non-frequent customer  
18  Non-frequent customer  
19  Non-frequent customer

In [47]:
ords_prods_merged_NV [['user_id', 'days_since_prior_order','median_days_between_orders','order_frequency_flag',]].tail(20)

user_id  days_since_prior_order  median_days_between_orders  \
32404839   206209                    18.0                        22.0   
32404840   206209                    18.0                        22.0   
32404841   206209                    18.0                        22.0   
32404842   206209                    18.0                        22.0   
32404843   206209                    18.0                        22.0   
32404844   206209                    18.0                        22.0   
32404845   206209                    18.0                        22.0   
32404846   206209                    18.0                        22.0   
32404847   206209                    18.0                        22.0   
32404848   206209                    18.0                        22.0   
32404849   206209                    18.0                        22.0   
32404850   206209                     7.0                        22.0   
32404851   206209                     7.0                        22.0   
32404852   206209                     7.0                        22.0   
32404853   206209                     7.0                        22.0   
32404854   206209                     7.0                        22.0   
32404855   206209                     7.0                        22.0   
32404856   206209                     7.0                        22.0   
32404857   206209                     7.0                        22.0   
32404858   206209                     7.0                        22.0   

           order_frequency_flag  
32404839  Non-frequent customer  
32404840  Non-frequent customer  
32404841  Non-frequent customer  
32404842  Non-frequent customer  
32404843  Non-frequent customer  
32404844  Non-frequent customer  
32404845  Non-frequent customer  
32404846  Non-frequent customer  
32404847  Non-frequent customer  
32404848  Non-frequent customer  
32404849  Non-frequent customer  
32404850  Non-frequent customer  
32404851  Non-frequent customer  
32404852  Non-frequent customer  
32404853  Non-frequent customer  
32404854  Non-frequent customer  
32404855  Non-frequent customer  
32404856  Non-frequent customer  
32404857  Non-frequent customer  
32404858  Non-frequent customer

In [49]:
(ords_prods_merged_NV.iloc[200000:200050])

Unnamed: 0.1  Unnamed: 0  order_id  user_id  order_number  \
200000         21842       21842   2342522     1365            23   
200001         21842       21842   2342522     1365            23   
200002         21842       21842   2342522     1365            23   
200003         21842       21842   2342522     1365            23   
200004         21842       21842   2342522     1365            23   
200005         21843       21843   1359003     1365            24   
200006         21843       21843   1359003     1365            24   
200007         21843       21843   1359003     1365            24   
200008         21843       21843   1359003     1365            24   
200009         21843       21843   1359003     1365            24   
200010         21843       21843   1359003     1365            24   
200011         21843       21843   1359003     1365            24   
200012         21843       21843   1359003     1365            24   
200013         21843       21843   1359003     1365            24   
200014         21844       21844     84632     1365            25   
200015         21844       21844     84632     1365            25   
200016         21844       21844     84632     1365            25   
200017         21844       21844     84632     1365            25   
200018         21844       21844     84632     1365            25   
200019         21844       21844     84632     1365            25   
200020         21844       21844     84632     1365            25   
200021         21845       21845   2624667     1365            26   
200022         21845       21845   2624667     1365            26   
200023         21845       21845   2624667     1365            26   
200024         21845       21845   2624667     1365            26   
200025         21845       21845   2624667     1365            26   
200026         21845       21845   2624667     1365            26   
200027         21845       21845   2624667     1365            26   
200028         21845       21845   2624667     1365            26   
200029         21845       21845   2624667     1365            26   
200030         21845       21845   2624667     1365            26   
200031         21845       21845   2624667     1365            26   
200032         21845       21845   2624667     1365            26   
200033         21845       21845   2624667     1365            26   
200034         21845       21845   2624667     1365            26   
200035         21845       21845   2624667     1365            26   
200036         21845       21845   2624667     1365            26   
200037         21845       21845   2624667     1365            26   
200038         21845       21845   2624667     1365            26   
200039         21845       21845   2624667     1365            26   
200040         21845       21845   2624667     1365            26   
200041         21846       21846    218831     1365            27   
200042         21846       21846    218831     1365            27   
200043         21846       21846    218831     1365            27   
200044         21846       21846    218831     1365            27   
200045         21846       21846    218831     1365            27   
200046         21846       21846    218831     1365            27   
200047         21846       21846    218831     1365            27   
200048         21846       21846    218831     1365            27   
200049         21846       21846    218831     1365            27   

        orders_day_of_week  orders_hour_of_day  days_since_prior_order  \
200000                   1                   9                     8.0   
200001                   1                   9                     8.0   
200002                   1                   9                     8.0   
200003                   1                   9                     8.0   
200004                   1                   9                     8.0   
200005                   4                  

In [52]:
ords_prods_merged_NV.loc[ords_prods_merged_NV['prices'] >100, 'prices'] = np.nan

In [54]:
ords_prods_merged_NV['prices'].max()

25.0

In [51]:
#exporting the dataframe
ords_prods_merged_NV.to_pickle(os.path.join(path,'02 Data','Prepared Data', 'ords_prods_merged_NV_Grouped.pkl'))